### PR TITLE
fix: rich text support related issues

### PR DESCRIPTION
**Pull request by whit3st**
## Description

Rich text support improvements for the template editor:
- Fix `[object Object]` display in migration dialogs — rich-text docs now show "Rich text (inline)" / "Rich text (block)" instead of raw JSON
- Fix false TYPE_MISMATCH in migration detection for `$ref` fields (e.g. rich-text properties) where `propSchema.type` is `undefined`, causing metadata-only changes (required↔optional) to trigger spurious migrations
- Fix rich-text example inputs overflowing in the data contract editor
- Fix ProseMirror double border with parent element
- Fix label click not focusing ProseMirror editor in data examples (custom elements aren't labelable per HTML spec)
- Polish the Rich Text Block UI: title-case label, show bound variable name in canvas block header, style the empty state with design tokens
- Show actual preview of variable, instead of hardcoded `[rich text value]` string
- Fix overflowing text in rich text block via CSS

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Chore (dependencies, CI, tooling)

## Component(s) Affected

- [ ] Backend (Spring Boot/Kotlin)
- [x] Frontend (Editor/TypeScript)
- [ ] Documentation
- [ ] CI/CD

## Checklist

- [x] My code follows the project's code style
- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or my feature works
- [x] New and existing tests pass locally
- [ ] I have updated the documentation if needed
- [ ] I have updated the CHANGELOG.md if this is a notable change
- [x] My commits follow [Conventional Commits](https://www.conventionalcommits.org/)

## Additional Notes

- `formatValue` in `MigrationAssistant` now uses `classifyValue` from the ref-type registry as the single source of truth for identifying rich-text doc shapes
- `detectMigrations` delegates to `shallowShapeCheck` for `$ref` properties, avoiding false TYPE_MISMATCH when `propSchema.type` is `undefined`
- Added comprehensive test coverage: `MigrationAssistant.test.ts` (32 tests, formatValue + DOM rendering) and `SchemaMigrationComprehensive.test.ts` (31 tests, migration detection edge cases)
- Empty state styling moved to a proper `richtextvar.css` file with `@layer components`, using design tokens

### CODE REVIEW

**Comment by whit3st** [comment]
Forgot to fix displayed expression/variable value, don't merge this yet.

image

File: modules/design-system/base.css [hunk 1]
```diff
@@ -34,7 +34,7 @@
   }
 
   /* Global focus-visible ring */
-  :focus-visible {
+  :focus-visible:not(.ProseMirror) {
     outline: none;
```
Comment (whit3st) [inline modules/design-system/base.css:38]: Before and after:

Image

Image

File: modules/editor/src/main/typescript/components/richTextVariable/richtextvar-registration.ts [hunk 2]
```diff
@@ -12,7 +12,11 @@ import './EpistolaRichTextVariablePreview.js';
 export function createRichTextVariableDefinition(): ComponentDefinition {
   return {
     type: 'richTextVariable',
-    label: 'Rich text block',
+    label: 'Rich Text Block',
+    getLabel: (node, _eng) => {
+      const binding = (node.props?.binding as string | undefined) ?? '';
+      return binding ? `Rich Text Block — ${binding}` : 'Rich Text Block';
+    },
```
Comment (whit3st) [inline modules/editor/src/main/typescript/components/richTextVariable/richtextvar-registration.ts:19]: No need to go into inspector and see which variable was selected

Image

File: modules/editor/src/main/typescript/data-contract/sections/ExampleForm.ts [hunk 3]
```diff
@@ -358,7 +358,15 @@ function renderFormField(
...
-          ${label}
+          <span
+            class="dc-tree-label"
+            role="label"
+            @click=${() => document.getElementById(fieldId)?.focus()}
+          >
+            ${name}${isRequired
+              ? html`<span class="dc-required-mark" aria-hidden="true">*</span>`
+              : nothing}
+          </span>
```
Comment (whit3st) [inline modules/editor/src/main/typescript/data-contract/sections/ExampleForm.ts:369]: This is for label not focusing the prose mirror content editable field. I chose programmatic focus for this with `role=label` which makes accessibility pass too

File: modules/editor/src/main/typescript/data-contract/binding-compatibility.ts [hunk 4]
```diff
@@ -57,13 +57,31 @@ export function formatFieldPathTypeLabel(fp: FieldPath): string {
...
+    const texts: string[] = [];
+    for (const node of b.content) {
+      if (!node || typeof node !== 'object') continue;
+      const n = node as { type?: string; text?: string };
+      if (n.type === 'text' && n.text) texts.push(n.text);
+      else if (n.type === 'hard_break') texts.push(' ');
+    }
+    if (texts.length > 0) paragraphs.push(texts.join(''));
+  }
+  return paragraphs.length > 0 ? paragraphs.join(' ') : '(empty)';
```
Comment (whit3st) [inline modules/editor/src/main/typescript/data-contract/binding-compatibility.ts:86]: This looks a bit ugly but I think word 'Preview' in the editor should preview the actual value🙃

**Comment by whit3st** [comment]
This is ready to be reviewed now @sdegroot
